db/sql/00_msar.sql

```diff
@@ -3737,3 +3738,41 @@ SELECT msar.list_records_from_table(
   null
 )
 $$ LANGUAGE SQL STABLE RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.delete_records_from_table(tab_id oid, rec_ids jsonb) RETURNS integer AS $$/*
+Delete records from table by id.
+
+Args:
+  tab_id: The OID of the table whose record we'll delete.
+  rec_ids: An array of primary key values
+
+The table must have a single primary key column.
+*/
+DECLARE
+  num_deleted integer;
+BEGIN
+  EXECUTE format(
+    $d$
+    WITH delete_cte AS (DELETE FROM %1$I.%2$I %3$s RETURNING *)
+    SELECT count(1) FROM delete_cte
+    $d$,
+    msar.get_relation_schema_name(tab_id),
+    msar.get_relation_name(tab_id),
+    msar.build_where_clause(
+      tab_id, jsonb_build_object(
+        'type', 'json_array_contains', 'args', jsonb_build_array(
+          jsonb_build_object('type', 'literal', 'value', rec_ids),
+          jsonb_build_object(
+            'type', 'convert_to_json', 'args', jsonb_build_array(
+              jsonb_build_object('type', 'attnum', 'value', msar.get_pk_column(tab_id))
+            )
+          )
+        )
+      )
+    )
+  ) INTO num_deleted;
+  RETURN num_deleted;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
```

Comment:
Would there be any performance benefits by converting this function to SQL? I guess we couldn't make it `STABLE` since it _does_ modify the database.

Reply:
we actually can't change this to an SQL function, since there wouldn't be a way to get the schema, table, or pkey column names. The overall flow of "build a string of the SQL to execute, then run it" doesn't work in SQL functions.